test/e2e/storage/vsphere/vsphere_utils.go

```diff
@@ -753,13 +753,15 @@ func getUUIDFromProviderID(providerID string) string {
 
 // GetReadySchedulableNodeInfos returns NodeInfo objects for all nodes with Ready and schedulable state
 func GetReadySchedulableNodeInfos(ctx context.Context, c clientset.Interface) []*NodeInfo {
-	nodeList, err := e2enode.GetReadySchedulableNodes(ctx, c)
-	framework.ExpectNoError(err)
 	var nodesInfo []*NodeInfo
-	for _, node := range nodeList.Items {
-		nodeInfo := TestContext.NodeMapper.GetNodeInfo(node.Name)
-		if nodeInfo != nil {
-			nodesInfo = append(nodesInfo, nodeInfo)
+	if TestContext.NodeMapper != nil {
+		nodeList, err := e2enode.GetReadySchedulableNodes(ctx, c)
+		framework.ExpectNoError(err)
+		for _, node := range nodeList.Items {
+			nodeInfo := TestContext.NodeMapper.GetNodeInfo(node.Name)
+			if nodeInfo != nil {
```

Comment:
Have you considered initializing `TestContext.NodeMapper` instead? e.g.

```go
if TestContext.NodeMapper != nil {
  TestContext.NodeMapper = NewNodeMapper()
}

// rest of the code
```

A concern I have is that `GetReadySchedulableNodeInfos()` might return a different result with the code proposed e.g. empty instead of possible nodeInfos from all schedulable nodes

Reply:
Dezhou, is this comment resolved already?

Reply:
Yes, just submitted the comment. I drafted but forgot to submit the reply.

Reply:
makes sense, thanks